diffrax/brownian/path.py

```diff
@@ -29,11 +37,28 @@ class UnsafeBrownianPath(AbstractBrownianPath):
     correlation structure isn't needed.)
     """
 
-    shape: Tuple[int] = eqx.static_field()
+    shape: PyTree[jax.ShapeDtypeStruct] = eqx.static_field()
     # Handled as a string because PRNGKey is actually a function, not a class, which
     # makes it appearly badly in autogenerated documentation.
     key: "jax.random.PRNGKey"  # noqa: F821
 
+    def __init__(
+        self,
+        shape: Union[Tuple[int, ...], PyTree[jax.ShapeDtypeStruct]],
+        key: "jax.random.PRNGKey",
+    ):
+        self.shape = (
+            jax.ShapeDtypeStruct(shape, None) if is_tuple_of_ints(shape) else shape
+        )
+        error_if(
+            not jtu.tree_all(
+                jtu.tree_map(lambda x: jnp.issubdtype(x.dtype, jnp.inexact), self.shape)
+            ),
+            "UnsafeBrownianPath dtypes all have to be floating-point.",
+        )
```

Comment:
This should just be a regular `ValueError`. The `error_if` function is for raising runtime errors, but this is an error we'd like to catch at trace time.

I.e.
```python
if any(not jnp.issubdtype(x.dtype, jnp.inexact) for x in jtu.tree_leaves(self.shape)):
     raise ValueError(...)
```

Reply:
That makes sense -- I've changed this